packages/storage-plugin/src/symbols.ts

```diff
@@ -58,12 +59,26 @@ export const NGXS_STORAGE_PLUGIN_OPTIONS = new InjectionToken('NGXS_STORAGE_PLUG
 
 export const STORAGE_ENGINE = new InjectionToken('STORAGE_ENGINE');
 
+/**
+ * StorageEngine API is similar to the WebStorage API (localStorage or sessionStorage) but can be used with asynchronous
+ * storage engines such as IndexedDB
+ */
 export interface StorageEngine {
-  readonly length: number;
-
-  getItem(key): any;
+  getItem(key): Observable<any>;
   setItem(key, val): void;
-  removeItem(key): void;
```

Comment:
Return back

Reply:
I removed som methods from ``StorageEngine`` since they were unused and would have forced people to make a more complex than necessary custom storage engine implementation.

Is there any plans to use theses methods in the future ?
Do you me to add back all of them or onlye ``removeItem`` ?

Reply:
in your case, these are critical changes and those who use it can break their application.